pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/IntermediateResultsBlock.java

```diff
@@ -80,9 +81,8 @@ public IntermediateResultsBlock() {
   /**
    * Constructor for selection result.
    */
-  public IntermediateResultsBlock(DataSchema dataSchema, Collection<Object[]> selectionResult) {
+  public IntermediateResultsBlock(DataSchema dataSchema) {
```

Comment:
The change in this class can be simplified by adding a `@Nullable Comparator<? super Object[]> comparator`, then we may add a new method `public PriorityQueue<Object[]> getSelectionResultAsPriorityQueue()`. I feel the current change is a little bit too complicated, and we might not want to extend this block

Reply:
I understand that. The change is quite large, but it provides type safety at compile time. For example, now it is clear that `MinMaxValueBasedSelectionOrderByCombineOperator` or `SelectionOrderByOperator` require priority queues.

Anyway, I understand that this change is the most subjective one, so I'm open to change it